spinnman/messages/eieio/create_eieio_data.py

```diff
@@ -118,145 +118,146 @@
 
```

Comment:
The whole EIEIO interface as I see it in SpiNNman is incredibly confusing and 
obscure. I agree wholeheartedly that this needs to be changed. Ideally there 
should be a single straightforward function for assembling the packet headers. I 
think there is also a proliferation of unnecessary types here. However given 
that the protocol itself at this level is going to change somewhat, with new 
packet encodings, implementation of this fix should wait until the new protocol 
- which is the obvious opportunity in any case to revamp the interface.

On 11/08/15 11:46, Alan Stokes wrote:
> In spinnman/messages/eieio/create_eieio_data.py
> <https://github.com/SpiNNakerManchester/SpiNNMan/pull/34#discussion_r36731931>:
>
>> @@ -118,145 +118,146 @@
>>
>
> This whole class seems wholely more complex than it needs to be, and feels very
> unscalable and unmanageable. To top it off, it seems it is missing certain
> classes of EIEIO data messages. There must be a cleaner way to interprete the
> EIEIO data packet interface. Though because this is rather tied into the
> buffered in functionality (and likely the buffered out stuff), im going to leave
> it for a future refactor. but OWCH!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpiNNakerManchester/SpiNNMan/pull/34/files#r36731931>.
>